roles/v2/mining-proxy/src/lib/downstream_mining.rs

```diff
@@ -419,22 +415,30 @@ impl
         &mut self,
         m: SubmitSharesStandard,
     ) -> Result<SendTo<UpstreamMiningNode>, Error> {
+        // TODO maybe we want to check if shares meet target before
+        // sending them upstream If that is the case it should be
+        // done by GroupChannel not here
         match &self.status {
             DownstreamMiningNodeStatus::Initializing => todo!(),
             DownstreamMiningNodeStatus::Paired(_) => todo!(),
             DownstreamMiningNodeStatus::ChannelOpened(Channel::DowntreamHomUpstreamGroup {
                 ..
             }) => {
                 let remote = self.upstream.as_ref().unwrap();
-                // TODO maybe we want to check if shares meet target before
-                // sending them upstream If that is the case it should be
-                // done by GroupChannel not here
                 let message = Mining::SubmitSharesStandard(m);
                 Ok(SendTo::RelayNewMessageToRemote(remote.clone(), message))
             }
+            DownstreamMiningNodeStatus::ChannelOpened(Channel::DowntreamHomUpstreamExtended {
+                ..
+            }) => {
+                let remote = self.upstream.as_ref().unwrap();
+                let res = UpstreamMiningNode::handle_std_shr(remote.clone(), m).unwrap();
```

Comment:
Is this unwrap safe? If it is and it's obvious, it might just be my limited knowing being newer to the code base... but otherwise, a comment explaining why could be helpful.